src/test/scala/com/fulcrumgenomics/illumina/SampleTest.scala

```diff
@@ -54,6 +54,22 @@ class SampleTest extends UnitSpec with OptionValues{
   }
 
   it should "support throw an exception if extended attribute keys are not all uppercase" in {
-    an[Exception] should be thrownBy new Sample(0, "ID", "NAME", extendedAttributes=Seq(("foo", "1"), ("bar", "2  ")).toMap)
+    an[Exception] should be thrownBy new Sample(0, "ID", "NAME", "LIBRARY", extendedAttributes=Seq(("foo", "1"), ("bar", "2  ")).toMap)
+  }
+
+  "Sample.hashCode" should "be the hash code of the concatenated barcode sequences" in {
```

Comment:
I think it would be better to write these tests to test the desired behaviour (e.g. hashCodes don't result in collisions for different samples) rather than the specific implementation.